gnocchi/storage/ceph.py

```diff
@@ -183,12 +183,17 @@ def _build_unaggregated_timeserie_path(metric, version):
         return (('gnocchi_%s_none' % metric.id)
                 + ("_v%s" % version if version else ""))
 
-    def _get_unaggregated_timeserie(self, metric, version=3):
+    def _get_or_create_unaggregated_timeseries_unbatched(
+            self, metric, version=3):
         try:
-            return self._get_object_content(
+            contents = self._get_object_content(
                 self._build_unaggregated_timeserie_path(metric, version))
         except rados.ObjectNotFound:
-            raise storage.MetricDoesNotExist(metric)
+            self._create_metric(metric)
+        else:
+            # _create_metric writes "" so replace it by None to indicate
+            # emptiness instead.
+            return contents or None
```

Comment:
curious, but why would contents be "". shouldn't every metric we create have something in or is this just a handling a case where it fails for some reason?

Reply:
Yes: if it fails to store for some reason, then _create_metric has written "" and then we need to return `None`. I agree that this should not happen if everything goes well :)